hbase-it/src/test/java/org/apache/hadoop/hbase/KubernetesClusterManager.java

```diff
@@ -0,0 +1,258 @@
+
+  @Override
+  public void start(ServiceType service, String hostname, int port) {
+    assertSupportedServiceType(service);
+    // presumably kubernetes is automatically starting pods, so nothing to do here.
+  }
+
+  @Override
+  public void stop(ServiceType service, String hostname, int port) throws IOException {
+    assertSupportedServiceType(service);
+    kill(hostname);
+  }
+
+  @Override
+  public void restart(ServiceType service, String hostname, int port) throws IOException {
+    assertSupportedServiceType(service);
+    kill(hostname);
+  }
+
+  @Override
+  public void kill(ServiceType service, String hostname, int port) throws IOException {
+    assertSupportedServiceType(service);
+    kill(hostname);
+  }
+
+  @Override
+  public void suspend(ServiceType service, String hostname, int port) {
+    throw unsupportedActionType("suspend");
+  }
+
+  @Override
+  public void resume(ServiceType service, String hostname, int port) {
+    throw unsupportedActionType("resume");
+  }
+
+  @Override
+  public boolean isRunning(ServiceType service, String hostname, int port) throws IOException {
+    final String roleSelector = roleSelectorForServiceType(service);
+    return isRunning(roleSelector, hostname);
+  }
+
+  private String roleSelectorForServiceType(final ServiceType serviceType) {
+    assertSupportedServiceType(serviceType);
+    switch (serviceType) {
+      case HADOOP_NAMENODE:
+        return getConf().get(NAMENODE_ROLE_SELECTOR_CONF_KEY, "namenode");
+      case HADOOP_DATANODE:
+        return getConf().get(DATANODE_ROLE_SELECTOR_CONF_KEY, "datanode");
+      case HADOOP_JOURNALNODE:
+        return getConf().get(JOURNALNODE_ROLE_SELECTOR_CONF_KEY, "journalnode");
+      case ZOOKEEPER_SERVER:
+        return getConf().get(ZOOKEEPER_ROLE_SELECTOR_CONF_KEY, "zookeeper");
+      case HBASE_MASTER:
+        return getConf().get(MASTER_ROLE_SELECTOR_CONF_KEY, "master");
+      case HBASE_REGIONSERVER:
+        return getConf().get(REGIONSERVER_ROLE_SELECTOR_CONF_KEY, "regionserver");
+      default:
+        throw new RuntimeException("should not happen");
+    }
+  }
+
+  private boolean isRunning(String roleSelector, String hostname) throws IOException {
+    if (api == null) {
+      return false;
+    }
+    final V1PodList list;
+    try {
+      list = api.listNamespacedPod(namespace, null, null, null, null,
+        "role in (" + roleSelector + ")", null, null, null, null, null);
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+    if (list == null) {
+      return false;
+    }
+    for (V1Pod item : list.getItems()) {
+      final String podName =
+        Optional.ofNullable(item).map(V1Pod::getMetadata).map(V1ObjectMeta::getName).orElse(null);
+      if (StringUtils.isEmpty(podName)) {
+        LOG.warn("Listing of namespace '{}' contains entry with empty pod name.", namespace);
+        continue;
+      }
+      if (hostname.startsWith(podName)) {
+        return true;
+      }
```

Comment:
Could you also check the Status and Readiness probes for the pod, rather than just existence of the pod?

I would assume you have some kind of readiness probe defined (e.g. one to check that the server is listening on its port is pretty easy to do) for the HBase processes.